File: src/gs/json-encode/JsonBenchmark.go
package main

import (
	"fmt"
	"encoding/json"
)

const ITERATIONS int = 1000;

var overallDuration float32 = 0.0;
var iterations int = ITERATIONS;

type Book struct {
	Title string
	Author string
	Price float32
	Pages int
}

var data [10]Book;

func execute(numberOfIterations int) {
	iterations = numberOfIterations;
	fmt.Printf("Iterations: %d \n\n", ITERATIONS);
	prepareData();

	for i := 0; i < iterations; i++ {
		jsonByte , err := json.Marshal(data);
		if (err != nil) {
			fmt.Println("Shit!");
			return;
		}
		jsonString := string(jsonByte[:]);
		//fmt.Println(jsonString);

		var jsonData [10]Book;
		err = json.Unmarshal([]byte(jsonString), &jsonData);
		if (err != nil) {
			fmt.Println("Shit!");
			return;
		}
		//fmt.Println(jsonData[3]);
	}
}

func getAverageDuration() float32 {
	return overallDuration / float32(iterations);
}

func prepareData() {
	data[0] = Book{
		"Der Herr der Ringe",
		"J.R.R. Tolkien",
		34.99,
		1245,
	};

	data[1] = Book{
		"Jagd auf Roter Oktober",
		"Tom Clancy",
		9.99,
		439,
	};

	data[2] = Book{
		"Ich, der Robot",
		"Isaac Asimov",
		10.99,
		247,
	};

	data[3] = Book{
		"Der Hundertjährige der aus dem Fenster stieg und verschwand",
		"Jonas Jonason",
		12.79,
		612,
	};

	data[4] = Book{
		"Cloud Atlas",
		"David Mitchell",
		19.99,
		567,
	};

	data[5] = Book{
		"Das Alphabethaus",
		"Jussi Adler Olsen",
		29.99,
		782,
	};

	data[6] = Book{
		"Eragon",
		"Christopher Paolini",
		34.99,
		987,
	};

	data[7] = Book{
		"1Q84",
		"Haruki Murakami",
		42.99,
		1439,
	};

	data[8] = Book{
		"Tintenherz",
		"Cornelia Funke",
		16.99,
		357,
	};

	data[9] = Book{
		"Wolfsblut",
		"Jack London",
		12.99,
		455,
	};
}

File: src/nodejs/json-encode/JsonBenchmark.js
/**
 * Created by lgaub on 10.12.2016.
 */
"use strict";

const ITERATIONS = 1000;

let prepareData = function () {
	let data = [];

	/** 1. */
	data.push({
		"title": "Der Herr der Ringe",
		"author": "J.R.R. Tolkien",
		"price": 34.99,
		"pages": 1245
	});

	/** 2. */
	data.push({
		"title": "Jagd auf Roter Oktober",
		"author": "Tom Clancy",
		"price": 9.99,
		"pages": 439
	});

	/** 3. */
	data.push({
		"title": "Ich, der Robot",
		"author": "Isaac Asimov",
		"price": 10.99,
		"pages": 247
	});

	/** 4. */
	data.push({
		"title": "Der Hundertjährige der aus dem Fenster stieg und verschwand",
		"author": "Jonas Jonason",
		"price": 12.79,
		"pages": 612
	});

	/** 5. */
	data.push({
		"title": "Cloud Atlas",
		"author": "David Mitchell",
		"price": 19.99,
		"pages": 567
	});

	/** 6. */
	data.push({
		"title": "Das Alphabethaus",
		"author": "Jussi Adler Olsen",
		"price": 29.99,
		"pages": 782
	});

	/** 7. */
	data.push({
		"title": "Eragon",
		"author": "Christopher Paolini",
		"price": 34.99,
		"pages": 987
	});

	/** 8. */
	data.push({
		"title": "1Q84",
		"author": "Haruki Murakami",
		"price": 42.99,
		"pages": 1439
	});

	/** 9. */
	data.push({
		"title": "Tintenherz",
		"author": "Cornelia Funke",
		"price": 16.99,
		"pages": 357
	});

	/** 10. */
	data.push({
		"title": "Wolfsblut",
		"author": "Jack London",
		"price": 12.99,
		"pages": 455
	});

	return data;
};

class JsonBenchmark {
	constructor () {
		this.iterations = ITERATIONS;
		this.overallDuration = 0;
	}

	/**
	 * Returns the average duration time for one encode and one decode operation.
	 * @returns {number}
	 */
	getAverageDuration() {
		return this.overallDuration / this.iterations;
	}

	/**
	 * Executes the benchmark.
	 * @param iterations[optional]
	 */
	execute(iterations = ITERATIONS) {
		let data = prepareData();
		let myString = "";
		let myJson = {};
		let start, stop;
		this.iterations = iterations;

		start = new Date().getTime();
		for (let i = 0; i < this.iterations; i++) {
			 myString = JSON.stringify(data);
			 myJson = JSON.parse(myString);
		}
		stop = new Date().getTime();
		this.overallDuration = stop - start;
	}
}

module.exports = JsonBenchmark;